src/component/requests/request.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import FullUrl from './full-url.js';
+import RequestLabels from './request-labels.js';
+import ContextMenu from '../context-menu/context-menu.js';
+
+const {func, object, number, bool} = React.PropTypes;
+
+export default class Request extends React.Component {
+
+  shouldComponentUpdate(nextProps) {
+    return this.props.request.id !== nextProps.request.id ||
+      this.props.positionTop !== nextProps.positionTop ||
+      this.props.isActive !== nextProps.isActive ||
+      this.props.isContextMenu !== nextProps.isContextMenu ||
+      this.done !== nextProps.request.done;
```

Comment:
This bit originally got me when I was moving around the context menu, and then just as I was about to go insane, I remembered about it.

Technically, `Request` and all of it's sub-components are now state-free, and there's minimal logic in `render()`, so it doesn't seem like a bad idea. I presume it was originally there because `Request` had state and was a bit more complex than it is now, so it was beneficial to check beforehand, multiplied by 10+ instances depending on window size.